x-pack/plugins/maps/public/shared/layers/vector_layer.js

```diff
@@ -233,28 +233,31 @@ export class VectorLayer extends AbstractLayer {
     try {
       const canSkip = await this._canSkipSourceUpdate(joinSource, sourceDataId, dataFilters);
       if (canSkip) {
+        const sourceDataRequest = this._findDataRequestForSource(sourceDataId);
+        const propertiesMap = sourceDataRequest ? sourceDataRequest.getData() : null;
         return {
-          shouldJoin: false,
-          join: join
+          dataHasChanged: false,
+          join: join,
+          propertiesMap: propertiesMap
         };
       }
       startLoading(sourceDataId, requestToken, dataFilters);
       const leftSourceName = await this.getSourceName();
       const {
-        rawData,
         propertiesMap
       } = await joinSource.getPropertiesMap(dataFilters, leftSourceName, join.getLeftFieldName());
-      stopLoading(sourceDataId, requestToken, rawData);
+      stopLoading(sourceDataId, requestToken, propertiesMap);
```

Comment:
track the `propertiesMap` iso the rawData. The advantage here is that is can be reused as-is.